### PR TITLE
IGNITE-19054 Use cmake conan wrapper for automatic conan configuration

### PR DESCRIPTION
**Pull request by ademakov**

### CODE REVIEW

File: modules/platforms/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -24,18 +24,61 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
+
+get_property(is_multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+
+# Install required libraries.
+if (is_multi_config)
+    # These settings are for multi-config generators like Visual Studio.
+    message(STATUS "Using a multi-config generator")
+
+    conan_cmake_configure(REQUIRES msgpack-c/4.0.0 gtest/1.12.1
+                          GENERATORS cmake_find_package_multi)
```
Comment (isapego) [inline modules/platforms/cpp/CMakeLists.txt:40]: Let's move a list of libraries to a separate variable.

**Review by isapego — CHANGES_REQUESTED** [review]

File: modules/platforms/cpp/CMakeLists.txt (hunk 1 above)
Comment (ademakov) [reply modules/platforms/cpp/CMakeLists.txt]: Done.

File: modules/platforms/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -23,19 +23,74 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
+option(WARNINGS_AS_ERRORS "Treat warning as errors" OFF)
+option(INSTALL_IGNITE_FILES "Install Ignite files" ON)
+
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
+get_property(is_multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+
+set(conan_packages msgpack-c/4.0.0 gtest/1.12.1)
```
Comment (isapego) [inline modules/platforms/cpp/CMakeLists.txt:40]: Let's keep it in a form "a library per line". This way it's easier to read and to track changes (i.e. with git).
```suggestion
set(conan_packages
    msgpack-c/4.0.0
    gtest/1.12.1
)
```

File: modules/platforms/cpp/DEVNOTES.md
Comment (isapego) [inline modules/platforms/cpp/DEVNOTES.md:1]: Maybe we should add instructions for building without Conan?

**Review by isapego — CHANGES_REQUESTED** [review]

**Review by isapego — APPROVED** [review]

**Pull request merged**